### PR TITLE
fixed 2 bugs and changed grid-system to generate real 960.gs (all classes)

### PR DESCRIPTION
**Pull request by ShPakvel**
Please consider my small changes for generating in "grid-system" mixin class ".container_XX" to simulate real 960.gs css framework.
However, there are 2 bugs fixed in this pull request. I think their solution will be useful at least.

Thanks for your gem. I've just started to look at haml+css+compass and your gem gave me good lesson for understanding difference between "for + include" and "enumerate + include".
I hope my changes will have sense for you. ;)

### CODE REVIEW

File: README.mkdn [hunk 1]
```diff
@@ -41,14 +41,13 @@ as defaults if the corresponding argument is omitted from a mixin call.
...
-To create a grid system that works like the original 960 Grid Sytem framework
+To create a grid system that works like the original 960 Grid System framework
 use the `+grid-system` mixin to generate the corresponding classes. You can
 also customize the number of columns as demonstrated in the following example.
 
 Example:
 
-    #wrap
-      +grid-system(24)
+    +grid-system(24)
```
Comment (nextmat) [inline README.mkdn:50]: This example will break as rewritten, the mixin needs to be nested under an id or class.

File: stylesheets/960/_grid.sass [hunk 2]
```diff
@@ -33,7 +33,7 @@ $ninesixty-class-separator: "_" !default
 
 =grids($cols: $ninesixty-columns, $gutter-width: $ninesixty-gutter-width)
   #{enumerate(".grid", 1, $cols, $ninesixty-class-separator)}
-    +grid-unit-base
+    +grid-unit-base($gutter-width)
```
Comment (nextmat) [inline stylesheets/960/_grid.sass:36]: This is already inherited by default if not set in grid-unit-base, so it doesn't need to be passed along here.

File: stylesheets/960/_grid.sass [hunk 3]
```diff
@@ -81,16 +81,17 @@ $ninesixty-class-separator: "_" !default
   #{enumerate(".push", 1, $cols, $ninesixty-class-separator)},
   #{enumerate(".pull", 1, $cols, $ninesixty-class-separator)}
     +grid-move-base
-  @for $n from 1 through $cols
+  @for $n from 1 through $cols - 1
```
Comment (nextmat) [inline stylesheets/960/_grid.sass:84]: Why did you change this to drop the last col?

File: templates/project/grid.sass [hunk 4]
```diff
@@ -19,14 +18,9 @@ $ninesixty-columns: 24
...
-    +grid-unit-base
   #header, #footer
-    +grid-width(24)
+    +grid(24)
   #sidebar
-    +grid-width(8)
+    +grid(8)
   #main-content
-    +grid-width(16)
+    +grid(16)
```
Comment (nextmat) [inline templates/project/grid.sass:26]: These changes remove the semantic use of the grid, if you are looking to generate an entire grid system see the examples before, for example:

```
.container_12
  +grid-system(12)
```

**Comment by nextmat** [comment]
Thanks for the patch. Some of the changes you made make me think you are trying to use the library in a different way than intended. Can you give me more details about what problems you are running into and trying to fix? 

I think you can get what you want without these changes, but if not we can work to get it sorted out and if so maybe we can improve the documentation to make it a bit clearer.

File: README.mkdn [hunk 5]
```diff
@@ -41,14 +41,13 @@ as defaults if the corresponding argument is omitted from a mixin call.
 Class-Based Grid System
 =======================
 
-To create a grid system that works like the original 960 Grid Sytem framework
+To create a grid system that works like the original 960 Grid System framework
```
Comment (nextmat) [inline README.mkdn:44]: Thanks for the catch, fixed.

File: README.mkdn (hunk 1 above)
Comment (ShPakvel) [inline README.mkdn:50]: Hi nextmat,
I think about +grid-system as mixin to generate the classes like in "the original 960 Grid System framework" (like it was described in 44 line above).
And I think it should generate all corresponding classes.
I am not sure that I understand real needs for mixin to be nested under an id or class. I think it is good to have all 960 classes independent on any ids and classes over it, just at the top-level like it is provided by original 960 css framework. And late we will able to use these classes in any many places on our page.
I think about it in such way also because I see like this plugin provides here two different approach of coding:
1) using mixins like +grid, +grid-prefix, etc. in sass file (later it will be just in css). So we just add corresponding properties in classes and ids we want, like #wrap.
2) using all classes, which people used to use from original 960 css framework, in haml file (later in html ;) ). And it is very convenient to have all these classes just including +grid-system to the top level (or even in a separate file).
So, when, for example, I would prefer to use more correct 1) approach ;) (thanks sass for his power),  some people will be able to use 2) approach with similar for them original 960 css framework with all classes they used to use.
Yep, I agree that because of it we have to write new code in different way. But I think it is not so big cost for beauty and logic of a bit new way. ;)
If you have another point of view, I would like to hear it and think about it.

File: templates/project/grid.sass (hunk 4 above)
Comment (ShPakvel) [inline templates/project/grid.sass:26]: please consider my thoughts above about it. ;)

File: stylesheets/960/_grid.sass (hunk 3 above)
Comment (ShPakvel) [inline stylesheets/960/_grid.sass:84]: because it is not necessary to pull or to push any grid (even grid_1) to col positions (col_12, col_16, etc.).
because then such grid_XX will be out of table (grid), then it will be before all columns or after all columns.
It is similar grid-prefix and grid-suffix.
You can look also at the original 960 css. There are only col-1 pulls, pushes, prefixes and suffixes.
But I've just noted that I forgot to put "- 1" to "enumerates" as well. I will do it later. ;)

File: stylesheets/960/_grid.sass (hunk 2 above)
Comment (ShPakvel) [inline stylesheets/960/_grid.sass:36]: If you use the +grids mixin separated of +grid-system and define your new $gutter-width parameter like the following:
+grids(16, 10)
then it will generate property "width" of grid accordingly gutter 10, but property "margin" of grid accordingly the default gutter 20.
if you do not use $gutter-width parameter in mixin ever, than I don't see the necessity of $gutter-width parameter in the mixin.

File: README.mkdn (hunk 1 above)
Comment (nextmat) [inline README.mkdn:50]: Yep, those are the exact two use cases we're covering. The reason that the mixin has to be indented is that sass requires it, i.e. a mixin has to mix into something, it can't stand alone. The example just limits the generated classes to be active inside a large #wrap div in this example, you could just as easily do:

```
body
  +grid-system(24)
```

If you want the exact functionality of the standard 960.gs, this will more or less give you that.

File: stylesheets/960/_grid.sass (hunk 2 above)
Comment (nextmat) [inline stylesheets/960/_grid.sass:36]: You're right, this is definitely a bug. I was looking at it wrong before.

File: stylesheets/960/_grid.sass (hunk 3 above)
Comment (nextmat) [inline stylesheets/960/_grid.sass:84]: Good point.

File: templates/project/grid.sass (hunk 4 above)
Comment (nextmat) [inline templates/project/grid.sass:26]: Yup, that's what I get for responding in a hurry. After the recent refactoring all of this can be simplified, good catch.

File: templates/project/grid.sass [hunk 6]
```diff
@@ -7,11 +7,10 @@
...
 
 // The following generates the default grids provided by the css version of 960.gs
-.container_12
-  +grid-system(12)
 
-.container_16
-  +grid-system(16)
++grid-system(12)
+
++grid-system(16)
```
Comment (nextmat) [inline templates/project/grid.sass:13]: Not sure I'm on board with this change to move the container class into the mixin. Kind of defeats the purpose. 

Users who want the default can simply retain the grid they prefer, with your changes you lose the ability to define the container div of your preference to generate the grid in. The power of sass mixins is granular use, so while perfect replication of the base 960.gs is certainly possible if you use on of the above it isn't necessarily the primary point.

File: stylesheets/960/_grid.sass [hunk 7]
```diff
@@ -81,16 +81,17 @@ $ninesixty-class-separator: "_" !default
...
-  +grid-suffixes($cols)
-  +grid-children
-  +grid-movements($cols)
+  .container#{$ninesixty-class-separator}#{$cols}
+    +grid-container
+    +grids($cols)
+    +grid-prefixes($cols)
+    +grid-suffixes($cols)
+    +grid-children
+    +grid-movements($cols)
```
Comment (nextmat) [inline stylesheets/960/_grid.sass:97]: See my comment on grid.sass, line 13

**Comment by nextmat** [comment]
Lots of good stuff in here, I made some errors when I looked at it first. Apologies for that.

I'm not convinced at all about moving the containers into the mixin. Consider that that will 1) break behavior for existing users 2) doesn't significantly improve the use case and 3) actually limits flexibility of the mixin somewhat.

If you can pull that stuff I'd be happy to take the rest of the patch. If you want to try to convince me, you can split the patch into two and we can continue discussion on that by itself.

**Comment by nextmat** [comment]
Thanks for including separate commits in the patch. I've accepted all the other parts, will close this pull request. If you want to talk further about the container issue, message me. Thanks for the contribution!
